pkg/operator/configobservation/authentication/apiaudiences.go

```diff
@@ -0,0 +1,107 @@
+package auth
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator/configobservation"
+)
+
+var (
+	audiencesPath = []string{"apiServerArguments", "api-audiences"}
+)
+
+// ObserveAPIAudiences changes apiServerArguments.api-audiences from
+// the default value if Authentication.Spec.ServiceAccountIssuer specifies a valid
+// non-empty value.
+func ObserveAPIAudiences(
+	genericListers configobserver.Listers,
+	recorder events.Recorder,
+	existingConfig map[string]interface{},
+) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
```

Comment:
is this safe?

Reply:
yes, the programmer is responsible for it to be correct, it's used as such in a few other spots as well